dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/basicToolApp/BasicTools.java

```diff
@@ -185,4 +189,15 @@ public String missingTitle(@ToolArg(name = "input", description = "input string"
         return input;
     }
 
+    @Tool(name = "cancellationTool", title = "Cancellable tool", description = "A tool that waits to be cancelled")
+    public String cancellationTool(Cancellation cancellation) throws InterruptedException {
+        int counter = 0;
+        while (counter++ < 5) {
```

Comment:
Please increase the number of iterations we wait so that the max wait time is at least 10 seconds to avoid spurious failures in the build.

In practise, if everything is working, we should never wait that long since the request will be cancelled immediately.